src/diffusers/models/autoencoder_kl.py

```diff
@@ -183,3 +183,12 @@ def forward(
             return (dec,)
 
         return DecoderOutput(sample=dec)
+
+
+class VAEDecoder(nn.Module):
+    def __init__(self, vae: AutoencoderKL):
+        super().__init__()
+        self.vae = vae
+
+    def forward(self, z: torch.FloatTensor, return_dict: bool = True):
+        return self.vae.decode(z, return_dict=return_dict)
```

Comment:
New module to trigger the previous hook, because it happens in the `pre_forward` method.

Unfortunately, I think we can't register `AutoencoderKL.decoder` instead, because the decode method performs a convolution before invoking the decoder: https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoder_kl.py#L123